rxmongo/src/main/scala/akka/contrib/persistence/mongodb/RxMongoSnapshotter.scala

```diff
@@ -44,9 +44,9 @@ class RxMongoSnapshotter(driver: RxMongoDriver) extends MongoPersistenceSnapshot
 
     for {
       s <- snaps(pid)
-      wr <- s.remove(BSONDocument(criteria: _*), writeConcern)
+      wr <- s.delete(true, writeConcern).one(BSONDocument(criteria: _*))
     } yield {
-      if (driver.useSuffixedCollectionNames && driver.suffixDropEmpty && wr.ok)
```

Comment:
Write operation never return a `Future.successful[WriteResult]` if not ok (and as already in `yield` ...)